cli/internal/cloudcmd/validators.go

```diff
@@ -81,44 +100,58 @@ func (v *Validator) updatePCR(pcrIndex uint32, encoded string) error {
-		WarnOnly: v.pcrs[pcrIndex].WarnOnly,
+	// new_measurement_value := hash(old_measurement_value || data_to_extend)
+	// Since we use the TPM2_PCR_Event (TPM) / TDG.MR.RTMR.EXTEND (TDX) call to extend the register, data_to_extend is the hash of our input
+	var expectedMeasurement []byte
+	switch v.attestationVariant {
+	case oid.AWSNitroTPM{}, oid.AzureTrustedLaunch{}, oid.AzureSEVSNP{}, oid.GCPSEVES{}, oid.QEMUVTPM{}:
+		hashedInput := sha256.Sum256(decoded)
+		oldExpected := v.measurements[measurementIndex].Expected
+		expectedMeasurementSum := sha256.Sum256(append(oldExpected[:], hashedInput[:]...))
+		expectedMeasurement = expectedMeasurementSum[:]
+	case oid.QEMUTDX{}:
+		hashedInput := sha512.Sum384(decoded)
+		oldExpected := v.measurements[measurementIndex].Expected
+		expectedMeasurementSum := sha512.Sum384(append(oldExpected[:], hashedInput[:]...))
+		expectedMeasurement = expectedMeasurementSum[:]
+	default:
+		return fmt.Errorf("updating measurement: unknown attestation variant")
+	}
+	v.measurements[measurementIndex] = measurements.Measurement{
+		Expected: expectedMeasurement,
+		WarnOnly: v.measurements[measurementIndex].WarnOnly,
 	}
 	return nil
 }
 
-func (v *Validator) setPCRs(config *config.Config) error {
+func (v *Validator) setMeasurements(config *config.Config) error {
 	switch v.attestationVariant {
 	case oid.AWSNitroTPM{}:
-		awsPCRs := config.Provider.AWS.Measurements
-		if len(awsPCRs) == 0 {
-			return errors.New("no expected measurement provided")
+		awsMeasurements := config.Provider.AWS.Measurements
+		if len(awsMeasurements) == 0 {
+			return errors.New("no expected AWS measurements provided")
 		}
-		v.pcrs = awsPCRs
+		v.measurements = awsMeasurements
 	case oid.AzureSEVSNP{}, oid.AzureTrustedLaunch{}:
-		azurePCRs := config.Provider.Azure.Measurements
-		if len(azurePCRs) == 0 {
-			return errors.New("no expected measurement provided")
+		azureMeasurements := config.Provider.Azure.Measurements
+		if len(azureMeasurements) == 0 {
+			return errors.New("no expected Azure measurements provided")
 		}
-		v.pcrs = azurePCRs
+		v.measurements = azureMeasurements
 	case oid.GCPSEVES{}:
-		gcpPCRs := config.Provider.GCP.Measurements
-		if len(gcpPCRs) == 0 {
-			return errors.New("no expected measurement provided")
+		gcpMeasurements := config.Provider.GCP.Measurements
+		if len(gcpMeasurements) == 0 {
+			return errors.New("no expected GCP measurements provided")
 		}
-		v.pcrs = gcpPCRs
-	case oid.QEMUVTPM{}:
-		qemuPCRs := config.Provider.QEMU.Measurements
-		if len(qemuPCRs) == 0 {
-			return errors.New("no expected measurement provided")
+		v.measurements = gcpMeasurements
+	case oid.QEMUVTPM{}, oid.QEMUTDX{}:
+		qemuMeasurements := config.Provider.QEMU.Measurements
+		if len(qemuMeasurements) == 0 {
+			return errors.New("no expected QEMU measurements provided")
 		}
-		v.pcrs = qemuPCRs
+		v.measurements = qemuMeasurements
+	default:
+		return errors.New("unknown attestation variant")
```

Comment:
```suggestion
		return fmt.Errorf("selecting measurements: unknown attestation variant: %s", v.attestationVariant.String())
```
Consistent error messages for the same error

Reply:
Yeah, you are right. Validator uses `oid.Getter` interface which doesn't implement `String()`.
This will be reworked in the future.
Should be fine to just return a simple error message without naming the variant (which shouldn't happen in the first place if the `NewValidator` constructor is used)